### DISCUSSION TITLE
Dynamic formschema not working properly

### DISCUSSION THREAD

**Discussion by joaoaraujo-hotmart** [formkit/formkit#652] [General]
Hi everyone, I'm using Formkit with FormKitSchemas and I'm facing issues for my behavior: I have a dynamic `schema` (using a computed) that changes your structure everytime user changes the country. If I have an input type `text` as the first field in a schema and I change to another schema where the first field is a `select` one, it still as a text field and do not update to a select properly. I notice that if I re-render the entire component everytime schema changes, it works well.. Do you think this is a schema reference bug or you don't think schema can be dynamic as I'm using?

**Comment by justin-schroeder** [comment]
First I should mention that swapping out schema is not recommended — its better to program reactivity *into* the schema. You can think of your schema like the `<template>` block of a `.vue` file — you cannot change what is in the block itself, but you can program reactivity into the block itself. That disclaimer aside, what you’re trying to do *should* work — but with so many variables at play it can be hard to identify precisely the issue. There is one issue open that could be related:

https://github.com/formkit/formkit/issues/610

But if you provide a simple playground reproduction I'm sure we can get further along the diagnosis.

**Reply by joaoaraujo-hotmart** [reply]
Thanks for your fast reply! Completely understand your point about the best way to do it, I'll plan a refactor to add the rectivity to the schema. The full schema swap was the way we was using on Formulate, I just replaced it with Formkit.
Anyway, here is a playground link with the same behavior: https://formkit.link/6b28fadcdb8f80e6355170655d0d10e4

**Reply by justin-schroeder** [reply]
Ahh, this is a case of Vue recycling DOM nodes. You can override this behavior by using an explicit `key`. We've also got a more general purpose fix for this odd behavior in the next significant release. Here’s an example of it working properly:

https://formkit.link/9bfd4081e4c6c497ed25a45a706ee3c0

And to your point — yes, this was a significant change from Vue Formulate where there was no reactivity in the schema itself so all reactivity had to be done via schema modification.
